packages/sanity/src/core/form/store/formState.ts

```diff
@@ -576,18 +576,21 @@ function prepareObjectInputState<T>(
         field,
         parent: parentProps,
         index,
-
         // the explicit type cast here is ok - we know that a fieldset can not have fieldsets
       }) as FieldMember | FieldError | null
+
+      if (fieldState?.kind === 'error') {
```

Comment:
This is also not strictly related to the fix, but the early return replaces the `as FieldMember` below